WeakAuras/BossMods.lua

```diff
@@ -0,0 +1,1598 @@
+      trigger.use_cloneId and "true" or "false",
+      trigger.use_extend and tonumber(trigger.extend or 0) or 0,
+      trigger.use_remaining and "true" or "false",
+      trigger.remaining and tonumber(trigger.remaining or 0) or 0,
+      trigger.use_cast == nil and "nil" or trigger.use_cast and "true" or "false",
+      trigger.use_isCooldown == nil and "nil" or trigger.use_isCooldown and "true" or "false",
+      trigger.remaining_operator or "<"
+    )
+  end,
+  statesParameter = "full",
+  args = {
+    {
+      name = "spellId",
+      display = L["Key"],
+      desc = L["The 'Key' value can be found in the BigWigs options of a specific spell"],
+      type = "string",
+      conditionType = "string",
+    },
+    {
+      name = "text",
+      display = L["Message"],
+      type = "longstring",
+      store = true,
+      conditionType = "string"
+    },
+    {
+      name = "remaining",
+      display = L["Remaining Time"],
+      type = "number",
+    },
+    {
+      name = "extend",
+      display = L["Offset Timer"],
+      type = "string",
+    },
+    {
+      name = "count",
+      display = L["Count"],
+      desc = L["Occurrence of the event, reset when aura is unloaded\nCan be a range of values\nCan have multiple values separated by a comma or a space\n\nExamples:\n2nd 5th and 6th events: 2, 5, 6\n2nd to 6th: 2-6\nevery 2 events: /2\nevery 3 events starting from 2nd: 2/3\nevery 3 events starting from 2nd and ending at 11th: 2-11/3\n\nOnly if BigWigs shows it on it's bar"],
+      type = "string",
+      store = true,
+      conditionType = "string",
+    },
+    {
+      name = "cast",
+      display = L["Cast Bar"],
+      desc = L["Filter messages with format <message>"],
+      type = "tristate",
+      test = "true",
+      init = "false",
+      conditionType = "bool"
+    },
+    {
+      name = "isCooldown",
+      display = L["Cooldown"],
+      desc = L["Cooldown bars show time before an ability is ready to be use, BigWigs prefix them with '~'"],
+      type = "tristate",
+      test = "true",
+      init = "false",
+      conditionType = "bool"
+    },
+    {
+      name = "cloneId",
+      display = L["Clone per Event"],
+      type = "toggle",
+      test = "true",
+      init = "false"
+    },
+  },
+  automaticrequired = true,
+}
+Private.category_event_prototype.addons["BigWigs Timer"] = L["BigWigs Timer"]
+
+-- Unified
+if BigWigsLoader or not DBM then
+  Private.ExecEnv.BossMods.Generic = Private.ExecEnv.BossMods.BigWigs
+  Private.ExecEnv.BossMods.BigWigs.isGeneric = true
+  Private.ExecEnv.BossMods.BigWigs.isInstalled = BigWigsLoader ~= nil
+elseif DBM then
+  Private.ExecEnv.BossMods.Generic = Private.ExecEnv.BossMods.DBM
+  Private.ExecEnv.BossMods.DBM.isGeneric = true
+  Private.ExecEnv.BossMods.DBM.isInstalled = true
+end
+
+-- make Private.ExecEnv.BossMods.Generic.GetStage public
+WeakAuras.GetBossStage = Private.ExecEnv.BossMods.Generic.GetStage
```

Comment:
There is a subtle error here. The BigWigs implementation requires a self parameter.